include/revng-c/RestructureCFG/CycleEquivalenceAnalysis.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+//
+// Copyright rev.ng Labs Srl. See LICENSE.md for details.
+//
+
+#include <map>
+
+#include "llvm/IR/CFG.h"
+#include "llvm/Pass.h"
+
+#include "revng/ADT/GenericGraph.h"
+
+#include "CycleEquivalenceClass.h"
+
+// `struct` used to represent the unique incremental `Index` that is
+// associated with each undirected edge in the `GenericGraph`
+struct EdgeIndex {
+private:
+  size_t Index = 0;
+
+public:
+  EdgeIndex(size_t Index) { this->Index = Index; }
+  size_t getValue() const { return Index; }
```

Comment:
Just `value()`.

Reply:
After the latest changes, an unique incremental identifier is part of the edge. Here https://github.com/revng/revng-c/blob/32e8b34c86eae349bd0b23530e69160e8d8b8b43/include/revng-c/RestructureCFG/CycleEquivalenceAnalysis.h#L44 is the field